packages/monitor-v2/src/monitor-polymarket/common.ts

```diff
@@ -374,52 +368,75 @@ export const getMarketsAncillary = async (
-        ancillaryData: string;
-        requestTimestamp: BigNumber;
-      }[];
-      // Update the ancillaryMap and results with the chunkResults
-      chunkResults.forEach((result, index) => {
-        const market = markets[i * chunkSize + index];
-        ancillaryMap.set(market.questionID, {
-          ancillaryData: result.ancillaryData,
-          requestTimestamp: result.requestTimestamp.toString(),
-        });
+  for (const adapterInfo of adapters) {
+    // Manually add polymarket abi to the abi decoder global so aggregateTransactionsAndCall will return the correctly decoded data.
+    decoder.abiDecoder.addABI(adapterInfo.abi);
+
+    const adapter = adapterInfo.contract;
+    const calls = markets
+      .filter((market) => sameAddress(market.resolvedBy, adapterInfo.address))
+      .map((market) => {
+        // In CTF v2, the questionId is the negRiskRequestID, not the questionID.
+        const isCtfV2Resolver = sameAddress(market.resolvedBy, params.ctfAdapterAddressV2);
+        const questionId = isCtfV2Resolver ? market.negRiskRequestID : market.questionID;
+        return {
+          target: adapter.address,
+          callData: adapter.interface.encodeFunctionData("questions", [questionId]),
+        };
       });
-    } catch (error) {
-      for (let j = 0; j < chunk.length; j++) {
-        const call = chunk[j];
-        const market = markets[i * chunkSize + j];
-        let ancillaryData, requestTimestamp;
-        try {
-          const adapter = sameAddress(market.resolvedBy, params.binaryAdapterAddress) ? binaryAdapter : ctfAdapter;
-          const result = await adapter.callStatic.questions(market.questionID);
-          ancillaryData = result.ancillaryData;
-          requestTimestamp = result.requestTimestamp.toString();
-        } catch {
-          ancillaryData = failed;
-          console.error("Failed to get ancillary data for market ", JSON.stringify(call), JSON.stringify(market));
-        }
-        ancillaryMap.set(market.questionID, {
-          ancillaryData,
-          requestTimestamp,
+
+    // batch call to multicall contract
+    const chunkSize = 25;
+    const chunks = [];
+
+    for (let i = 0; i < calls.length; i += chunkSize) {
+      chunks.push(calls.slice(i, i + chunkSize));
+    }
+
+    const chunkPromises = chunks.map(async (chunk, i) => {
+      try {
+        const chunkResults = (await aggregateTransactionsAndCall(multicall.address, web3, chunk)) as {
+          ancillaryData: string;
+          requestTimestamp: BigNumber;
+        }[];
+        // Update the ancillaryMap and results with the chunkResults
+        chunkResults.forEach((result, index) => {
+          const market = markets[i * chunkSize + index];
+          ancillaryMap.set(market.questionID, {
+            ancillaryData: result.ancillaryData,
+            requestTimestamp: result.requestTimestamp.toString(),
+          });
         });
+      } catch (error) {
```

Comment:
im not sure i understand this but is this saying if a multicall chunk fails, then try each item in chunk indpenendtly before finally giving up? why is it so important to optmize this with multicall?